WeakAuras/Features.lua

```diff
@@ -0,0 +1,116 @@
+---@type string, Private
+local addon, Private = ...
+
+---@alias BuildType "dev" | "pr" | "alpha" | "beta" | "release"
+
+---@class feature
+---@field id string
+---@field autoEnable? BuildType[]
+---@field requiredByAura? fun(self: self, aura: auraData): boolean
+---@field enabled? boolean
+---@field persist? true
+---@field sub? SubscribableObject
+
+---@class Features
+local Features = {
+  ---@type table<string, feature>
+  __feats = {}
+}
+Private.Features = Features
+
+---@param id string
+function Features:Exists(id)
+  return self.__feats[id] ~= nil
+end
+
+---@param id string
+function Features:Enabled(id)
+    return self:Exists(id) and self.__feats[id].enabled
+end
+
+---@param id string
+function Features:Enable(id)
+  if self:Exists(id) and not self.__feats[id].enabled then
+    self.__feats[id].enabled = true
+    if self.__feats[id].persist then
+      Private.db.features[id] = true
+    end
+    self.__feats[id].sub:Notify("ENABLED")
+  end
+end
```

Comment:
manually enable a feature may require a reload
```Lua
if WeakAuras.IsLoginFinished() and self.__feats[id].enableBeforeLoginFinished then
  -- called from /wa optin and feature needs to be enable earlier
  ReloadUI()
else
  self.__feats[id].sub:Notify("ENABLED")
end
```

Reply:
Hmm, that depends on how code which depends on a feature is written, surely? I'd 'prefer' that code 'magically' works without having to reload ui if possible, if nothing else than for my personal convenience.